### PR TITLE
Features/rhpm ur02

### PR DESCRIPTION
**Pull request by bharathkeerthi**
Forgot password screen of validation

### CODE REVIEW

File: web/src/app/Login/forgotpassword.component.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+@Component({
+    selector: 'app-forgotpassword',
+    templateUrl: 'forgotpassword.component.html',
+})
+export class ForgotpasswordComponent implements OnInit {
+
+    ForgotpasswordModel: UserViewModel = {} as UserViewModel;
+    constructor(
+        private forgotpasswordServiceApp: ForgotpasswordServiceApp, private location: Location,
+        private router: Router) {
```
Comment (salilsaini868) [inline web/src/app/Login/forgotpassword.component.ts]: what is use of location here ?

File: web/src/app/Login/login.component.html [hunk 2]
```diff
@@ -21,7 +21,7 @@
                                     </div>
                                 </div>
                                 <div class="pull-right forgot">
-                                    <a href="javascript:void(0)">Forgot password</a>
+                                    <a data-toggle="modal" (click)="navigate()">{{ 'USER.FORGOTPASSWORD' | translate }} </a>
```
Comment (salilsaini868) [inline web/src/app/Login/login.component.html]: give proper names to function

File: web/src/app/Login/shared/forgotpassword.serviceApp.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import 'rxjs/Rx';
+import { HttpClientModule, HttpClient } from '@angular/common/http';
+import { LoginService, RoleService } from '../../webapi/services';
+import { UserService } from '../../webapi/services';
+
+@Injectable()
+export class ForgotpasswordServiceApp {
+
+    constructor(private http: HttpClient, private apiForgotpasswordService: UserService, private apiRoleService: RoleService) { }
+
```
Comment (salilsaini868) [inline web/src/app/Login/shared/forgotpassword.serviceApp.ts]: why role service used here ?

File: web/src/app/Login/shared/login.module.ts [hunk 4]
```diff
@@ -23,9 +25,9 @@ import { SharedModule } from '../../shared/shared.module';
   exports: [
   ],
   declarations: [
-    LoginComponent
+    LoginComponent,ForgotpasswordComponent
   ],
-  providers: [LoginServiceApp],
+  providers: [LoginServiceApp, ForgotpasswordServiceApp],
```
Comment (salilsaini868) [inline web/src/app/Login/shared/login.module.ts]: use login service for Forgotpassword

**Review by salilsaini868 — CHANGES_REQUESTED** [review]

**Comment by salilsaini868** [comment]
resolve conflicts

File: web/src/app/Login/forgotpassword.component.ts (hunk 1 above)
Comment (bharathkeerthi) [reply web/src/app/Login/forgotpassword.component.ts]: done

File: web/src/app/Login/login.component.html (hunk 2 above)
Comment (bharathkeerthi) [reply web/src/app/Login/login.component.html]: done

File: web/src/app/Login/shared/forgotpassword.serviceApp.ts (hunk 3 above)
Comment (bharathkeerthi) [reply web/src/app/Login/shared/forgotpassword.serviceApp.ts]: done

File: web/src/app/Login/shared/login.module.ts (hunk 4 above)
Comment (bharathkeerthi) [reply web/src/app/Login/shared/login.module.ts]: done

File: web/src/app/Login/shared/login.serviceApp.ts [hunk 5]
```diff
@@ -4,13 +4,12 @@ import { Observable } from 'rxjs/Observable';
...
 import { LoginService, RoleService } from '../../webapi/services';
-
-
+import { UserService } from '../../webapi/services';
 
 @Injectable()
 export class LoginServiceApp {
 
-    constructor(private http: HttpClient, private apiLoginService: LoginService, private apiRoleService: RoleService) { }
+    constructor(private http: HttpClient, private apiLoginService: LoginService, private apiForgotpasswordService: UserService, private apiRoleService: RoleService) { }
```
Comment (salilsaini868) [inline web/src/app/Login/shared/login.serviceApp.ts]: remove them private apiForgotpasswordService: UserService, private apiRoleService: RoleService
or mover to corresponding serices

File: web/src/app/Login/shared/login.serviceApp.ts [hunk 6]
```diff
@@ -19,4 +18,7 @@ export class LoginServiceApp {
     getUser(): Observable<any> {
```
Comment (salilsaini868) [inline web/src/app/Login/shared/login.serviceApp.ts]: move this method to corresposnding service

**Review by salilsaini868 — CHANGES_REQUESTED** [review]

**Comment by salilsaini868** [comment]
resolve conflicts

**Review by salilsaini868 — APPROVED** [review]

**Pull request merged**